src/Diagnostics.ModelsAndUtils/Models/GeoMaster/AzureSupportCenterInsight.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Diagnostics.ModelsAndUtils.Attributes;
+using Diagnostics.ModelsAndUtils.Models.ResponseExtensions;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Models.GeoMaster
+{
+    public class AzureSupportCenterInsight
+    {
+        public Guid Id { get; set; }
+
+        public string Title { get; set; }
+
+        public ImportanceLevel ImportanceLevel { get; set; }
+
+        public string InsightFriendlyName { get; set; }
+
+        public string IssueCategory { get; set; }
+
+        public string IssueSubCategory { get; set; }
+
+        public Text Description { get; set; }
+
+        public RecommendedAction RecommendedAction { get; set; }
+
+        public CustomerReadyContent CustomerReadyContent { get; set; }
+
+        public IEnumerable<Link> Links { get; set; }
+
+        public string ConfidenceLevel { get; set; }
+
+        public string Scope { get; set; }
+    }
+
+    public enum ImportanceLevel
+    {
+        SomeLevel
+    }
+
+    public class CustomerReadyContent
+    {
+        public string ArticleId { get; set; }
+
+        public Text Text { get; set; }
+    }
+
+    public class Text
+    {
+        public bool IsMarkdown { get; set; }
+
+        public string Value { get; set; }
+    }
+
+    public class Link
+    {
+        public int Type { get; set; }
+
+        public string Text { get; set; }
+
+        public string Uri { get; set; }
+    }
+
+    public class RecommendedAction
+    {
+        public Guid Id { get; set; }
+
+        public Text Text { get; set; }
+    }
+
+    public static class AzureSupportCenterInsightUtilites
+    {
+        public static AzureSupportCenterInsight CreateInsight<TResource>(Insight insight, OperationContext<TResource> context, Definition detector)
+            where TResource : IResource
+        {
+
+            // Logic in line before would have to be updated if we added more resource types. 
+            // TODO: May make sense to have resource type enums contain the resource type string
```

Comment:
+1 on TODO. We want to minimize such conditions as it will be hard to onboard a new resource in future.